### PR TITLE
Implement alpha equivalence proofs

### PR DESCRIPTION
**Pull request by ajreynol**
This is a modified version of https://github.com/cvc5/cvc5/pull/6137 which accounts for extended rewriting between quantified formulas that are considered alpha equivalent.

It also generalizes the proof rule ALPHA_EQUIV.  Notice that if we were to make this rule more pedantic, we would check for variable shadowing during substitution, although this is not currently done.

### CODE REVIEW

File: src/proof/proof_rule.h [hunk 1]
```diff
@@ -835,14 +835,21 @@ enum class PfRule : uint32_t
   // has prefix "QUANTIFIERS_INST_E_MATCHING", then t is the trigger that
   // generated the instantiation.
   INSTANTIATE,
+  // ======== Alpha equivalence
+  // Children: none
+  // Arguments: (F, y1 = z1, ..., yn = zn)
```
Comment (HanielB) [inline src/proof/proof_rule.h:840]:
```suggestion
  // Arguments: (F, (= y1 z1), ..., (= yn zn)
```

File: src/theory/quantifiers/proof_checker.cpp [hunk 2]
```diff
@@ -120,6 +121,41 @@ Node QuantifiersProofRuleChecker::checkInternal(
...
+    {
+      if (args[i].getKind() != kind::EQUAL)
+      {
+        return Node::null();
+      }
+      for (size_t j = 0; j < 2; j++)
+      {
+        Node v = args[i][j];
+        if (v.getKind() != kind::BOUND_VARIABLE
+            || allVars[j].find(v) != allVars[j].end())
```
Comment (HanielB) [inline src/theory/quantifiers/proof_checker.cpp:146]: Feels a bit odd to use sets when you are enforcing uniqueness. Maybe just use vector?

**Review by HanielB — CHANGES_REQUESTED** [review]
Maybe add a regression, like that of #6137?

File: src/theory/quantifiers/proof_checker.cpp (hunk 2 above)
Comment (ajreynol) [reply src/theory/quantifiers/proof_checker.cpp:146]: I think set is cleaner in this case.

Comment (HanielB) [reply src/theory/quantifiers/proof_checker.cpp:146]: ok

**Review by HanielB — APPROVED** [review]

**Pull request merged**